src/components/MobileModal/MobileModal.tsx

```diff
@@ -0,0 +1,337 @@
+        (openFraction < 1 - threshold || averageSpeed > 0.5)
+      ) {
+        isAnimating = currentY.current - startY.current !== 0
+        modalRef.current.style.transition = `transform 0.4s`
+        modalRef.current.style.transform = `translateY(${
+          window.innerHeight - overflowHeight
+        }px)`
+        currentY.current = window.innerHeight - overflowHeight
+        isOpen = false
+      } else {
+        isAnimating = currentY.current - startY.current !== 0
+        modalRef.current.style.transition = `transform 0.4s`
+        modalRef.current.style.transform = `translateY(${
+          swipeState.isOpen ? topSpacing : window.innerHeight - overflowHeight
+        }px)`
+        currentY.current = swipeState.isOpen
+          ? topSpacing
+          : window.innerHeight - overflowHeight
+      }
+      setSwipeState({
+        ...swipeState,
+        isAnimating,
+        isDragging: false,
+        isOpen,
+      })
+    }
+  }
+
+  function handleTouchMove(event: TouchEvent<HTMLDivElement>): void {
+    event.stopPropagation()
+    endTime.current = new Date()
+
+    /* istanbul ignore else */
+    if (modalRef.current) {
+      const { clientY } = event.touches[0]
+      const diff = clientY - startY.current
+      const diffToTranslate =
+        (swipeState.isOpen ? topSpacing : window.innerHeight - overflowHeight) +
+        diff
+      const shouldTranslate =
+        diffToTranslate >= topSpacing &&
+        diffToTranslate <= window.innerHeight - overflowHeight
+      if (shouldTranslate)
+        modalRef.current.style.transform = `translateY(${diffToTranslate}px)`
+      currentY.current = clientY
+    }
+  }
+
+  function handleTouchStart(event: TouchEvent<HTMLDivElement>): void {
+    event.stopPropagation()
+    bodyRef.current.style.overflow = `hidden`
+    const { clientY } = event.touches[0]
+    currentY.current = clientY
+    startTime.current = new Date()
+    startY.current = clientY
+
+    setSwipeState({
+      ...swipeState,
+      isDragging: true,
+    })
+  }
+
+  function handleTransitionEnd(event: TransitionEvent<HTMLDivElement>): void {
+    if (event.target === modalRef.current) {
+      if (modalRef.current && draggable)
+        modalRef.current.style.transition = `none`
+      setSwipeState({
+        ...swipeState,
+        isAnimating: false,
+      })
+      if (!swipeState.isOpen && onClose) onClose()
+    }
+  }
+
+  return ReactDOM.createPortal(
+    <React.Fragment>
+      <div
+        data-testid="test-background"
+        onTouchEnd={handleDismiss}
+        ref={backgroundRef}
+        sx={{
+          backgroundColor: `text.0`,
+          display:
+            swipeState.isAnimating ||
+            swipeState.isDragging ||
+            swipeState.isOpen ||
+            open
+              ? `block`
+              : `none`,
+          height: `100vh`,
+          left: 0,
+          // Computed opacity to match our text.40
+          opacity: 0.6017699115044248,
```

Comment:
Can't you use `"text.alpha.40"` for the `backgroundColor` in this case? Or is the opacity important for other elements?

Reply:
Actually, the opacity property is required to maintain some visibility on the overlaid elements.

Reply:
Fixed